src/main/java/core/basesyntax/FileReaderImpl.java

```
@@ -0,0 +1,20 @@
+package core.basesyntax;
+
+import core.interfaces.FileReader;
+import java.io.IOException;
+import java.io.UncheckedIOException;
+import java.nio.file.Files;
+import java.nio.file.Paths;
+import java.util.List;
+
+public class FileReaderImpl implements FileReader {
+
+    @Override
+    public List<String> read(String path) {
+        try {
+            return Files.readAllLines(Paths.get(path));
+        } catch (IOException e) {
+            throw new UncheckedIOException(e);

```

The catch block rethrows `UncheckedIOException` with no informative message. The checklist says: "Don't ignore exceptions in try-catch blocks ... Let's rethrow a RuntimeException with an **informative** message and exception object." Replace `throw new UncheckedIOException(e);` with something like `throw new RuntimeException("Can't read file by path: " + path, e);` so callers receive an informative message and the original cause.